app/models/batch_entry.rb

```diff
@@ -12,11 +12,15 @@ def failed?
     state == "failed"
   end
 
+  def skipped?
+    state == "skipped"
+  end
+
   def complete?
-    failed? || succeeded?
+    failed? || succeeded? || skipped?
   end
 
   def check_job_complete
-    job.check_complete if job && complete?
+    job.check_complete if complete? && job
```

Comment:
This is an extremely minor performance improvement, in that it can check attributes before making a DB call.